templates/repo/diff/conversation.tmpl

```diff
@@ -1,21 +1,38 @@
 {{$resolved := (index .comments 0).IsResolved}}
+{{$invalid := (index .comments 0).Invalidated}}
 {{$resolveDoer := (index .comments 0).ResolveDoer}}
 {{$isNotPending := (not (eq (index .comments 0).Review.Type 0))}}
+{{$referenceUrl := printf "%s#%s" $.Issue.Link (index .comments 0).HashTag}}
 <div class="conversation-holder" data-path="{{(index .comments 0).TreePath}}" data-side="{{if lt (index .comments 0).Line 0}}left{{else}}right{{end}}" data-idx="{{(index .comments 0).UnsignedLine}}">
 	{{if $resolved}}
 		<div class="ui attached header resolved-placeholder gt-df gt-ac gt-sb">
-			<div class="ui grey text">
+			<div class="ui grey text gt-df gt-ac gt-fw gt-gap-2">
 				{{svg "octicon-check" 16 "icon gt-mr-2"}}
-				<b>{{$resolveDoer.Name}}</b> {{$.locale.Tr "repo.issues.review.resolved_by"}}
+				{{if $resolved}}
+					<b>{{$resolveDoer.Name}}</b> {{$.locale.Tr "repo.issues.review.resolved_by"}}
+				{{end}}
+				{{if $invalid}}
+					<a href="{{AppSubUrl}}{{$referenceUrl}}" class="ui label basic small gt-ml-3" data-tooltip-content="{{$.locale.Tr "repo.issues.review.outdated_description"}}">
+						{{$.locale.Tr "repo.issues.review.outdated"}}
+					</a>
+				{{end}}
 			</div>
-			<div>
-				<button id="show-outdated-{{(index .comments 0).ID}}" data-comment="{{(index .comments 0).ID}}" class="ui tiny labeled button show-outdated gt-df gt-ac">
+			<div class="gt-df gt-ac gt-gap-3">
+				<button id="show-outdated-{{(index .comments 0).ID}}" data-comment="{{(index .comments 0).ID}}" class="ui tiny labeled button show-outdated gt-df gt-ac{{if not $resolved}} gt-hidden{{end}}">
 					{{svg "octicon-unfold" 16 "gt-mr-3"}}
-					{{$.locale.Tr "repo.issues.review.show_resolved"}}
+					{{if $resolved}}
+						{{$.locale.Tr "repo.issues.review.show_resolved"}}
+					{{else}}
+						{{$.locale.Tr "repo.issues.review.show_outdated"}}
+					{{end}}
 				</button>
-				<button id="hide-outdated-{{(index .comments 0).ID}}" data-comment="{{(index .comments 0).ID}}" class="ui tiny labeled button hide-outdated gt-df gt-ac gt-hidden">
+				<button id="hide-outdated-{{(index .comments 0).ID}}" data-comment="{{(index .comments 0).ID}}" class="ui tiny labeled button hide-outdated gt-df gt-ac{{if $resolved}} gt-hidden{{end}}">
 					{{svg "octicon-fold" 16 "gt-mr-3"}}
-					{{$.locale.Tr "repo.issues.review.hide_resolved"}}
+					{{if $resolved}}
+						{{$.locale.Tr "repo.issues.review.hide_resolved"}}
+					{{else}}
+						{{$.locale.Tr "repo.issues.review.hide_outdated"}}
+					{{end}}
```

Comment:
These two buttons now look strange to me.

`{{if not $resolved}} gt-hidden{{end}}` + `{{if $resolved}}show resolved {{else}}show outdated {{end}}`

`{{if $resolved}} gt-hidden{{end}}` + `{{if $resolved}}hide resolved {{else}}hide outdated {{end}}`

Why the "outdated" is controlled by "resolved"?

Reply:
It's the same logic as the existing in Conversation tab (see https://github.com/go-gitea/gitea/blob/main/templates/repo/issue/view_content/comments.tmpl#L498 for the source of this logic ;) ) .

Reply:
I do not think they are the same. 

`comments.tmpl#L498` has another check ahead: `{{if or $invalid $resolved}}`

I don't see how `$invalid` is checked here.

Reply:
Good find as always - thanks @wxiaoguang . The logic here has changed during the review and probably the show outdated block is not needed anymore. I'll have a look at this probably today evening (german time).

Reply:
But apart of this - do you see any other blocking issues to get an LGTM?

Reply:
Overall LGTM, this is the last question in my mind.